### PR TITLE
fix: [PIE-6444]: multi svc envs as RT

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

When multi service and multi infra were marked runtime in template, it gets saved as runtime but on usage, it automatically sets to empty array

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts [hunk 1]
```diff
@@ -16,4 +16,5 @@ export interface DeployClusterEntityCustomStepProps {
 
 export interface DeployClusterEntityCustomInputStepProps extends DeployClusterEntityCustomStepProps {
   deployToAllClusters?: boolean
+  showEnvironmentsSelectionInputField?: boolean
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts:19]: Can we make this as required field?

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx [hunk 2]
```diff
@@ -46,6 +47,7 @@ export interface DeployEnvironmentEntityInputStepProps extends Required<DeployEn
     path?: string
     readonly?: boolean
   }
+  stepViewType?: StepViewType
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:50]: Same as above

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx [hunk 3]
```diff
@@ -202,29 +211,48 @@ export default function DeployEnvironmentEntityInputStep({
       const environmentObject: EnvironmentYamlV2 = Array.isArray(environmentsFormValues)
         ? environmentsFormValues.find((env: EnvironmentYamlV2) => env.environmentRef === envId)
         : {
-            ...(gitOpsEnabled ? { gitOpsClusters: [] } : { infrastructureDefinitions: [] })
+            ...(gitOpsEnabled
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:214]: Use of if else will make this code more readable

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx [hunk 4]
```diff
@@ -202,29 +211,48 @@ export default function DeployEnvironmentEntityInputStep({
...
+              : { infrastructureDefinitions: RUNTIME_INPUT_VALUE })
           }
 
       let environmentInputs = isMultiEnvironment
         ? environmentObject?.environmentInputs
         : get(formik.values, `${pathToEnvironments}environmentInputs`)
 
       if (!environmentInputs || isValueRuntimeInput(environmentInputs)) {
-        environmentInputs = envTemplateValue ? clearRuntimeInput(envTemplateValue) : undefined
+        environmentInputs = envTemplateValue
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:224]: Same as above

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx [hunk 5]
```diff
@@ -202,29 +211,48 @@ export default function DeployEnvironmentEntityInputStep({
...
+          ? deployToAllEnvironments && stepViewType === StepViewType.TemplateUsage
+            ? envTemplateValue
+            : clearRuntimeInput(envTemplateValue)
+          : undefined
       } else {
-        environmentInputs = merge(envTemplateValue ? clearRuntimeInput(envTemplateValue) : undefined, environmentInputs)
+        environmentInputs = merge(
+          envTemplateValue
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:231]: Same as above

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx [hunk 6]
```diff
@@ -217,10 +221,20 @@ export default function DeployInfrastructureEntityInputStep({
         : get(formik.values, localPath)
 
       if (!infrastructureInputs || isValueRuntimeInput(infrastructureInputs)) {
-        infrastructureInputs = infraTemplateValue ? clearRuntimeInput(infraTemplateValue) : undefined
+        infrastructureInputs = infraTemplateValue
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx:224]: Use of if else will make this code more readable

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx [hunk 7]
```diff
@@ -217,10 +221,20 @@ export default function DeployInfrastructureEntityInputStep({
...
+          ? // This condition helps to retain runtime value when any(pipeline/stage) template is used in pipeline
+            deployToAllInfrastructures && stepViewType === StepViewType.TemplateUsage
+            ? infraTemplateValue
+            : clearRuntimeInput(infraTemplateValue)
+          : undefined
       } else {
         infrastructureInputs = merge(
-          infraTemplateValue ? clearRuntimeInput(infraTemplateValue) : undefined,
+          infraTemplateValue
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx:232]: same as above

**Review by pankajsingh-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts:19]: No. This is not required when using single environment. This is required only for multi environments & environment group

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:50]: No. This is not required when using single environment. This is required only for multi environments & environment group

File: src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/types.ts:19]: In that case you can pass false value.

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx (hunk 3 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:214]: I'm thinking of a clean solution around it. This particular useEffect has a lot of conditions. Ternary operators make it compact and restrict variable conditions.

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx (hunk 4 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:224]: I'm thinking of a clean solution around it. This particular useEffect has a lot of conditions. Ternary operators make it compact and restrict variable conditions.

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx (hunk 5 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:231]: I'm thinking of a clean solution around it. This particular useEffect has a lot of conditions. Ternary operators make it compact and restrict variable conditions.

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx (hunk 6 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx:224]: I'm thinking of a clean solution around it. This particular useEffect has a lot of conditions. Ternary operators make it compact and restrict variable conditions.

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx (hunk 7 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployInfrastructureEntityStep/DeployInfrastructureEntityInputStep.tsx:232]: I'm thinking of a clean solution around it. This particular useEffect has a lot of conditions. Ternary operators make it compact and restrict variable conditions.

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**